### PR TITLE
Match index sorting

### PR DESCRIPTION
**Pull request by ytolstyk**

### CODE REVIEW

File: app/models/pool.rb [hunk 1]
```diff
@@ -117,4 +117,14 @@ def add_existing_fighter(tournament_id, fighter)
       "#{fighter.first_name} #{fighter.last_name} added"
     end
   end
+
+  def update_match_indices(matches_hash)
+    success = true
+    matches_hash.values.each do |match|
+      current_match = matches.find(match[:match_id])
+      success = false if !current_match.update(index: match[:index])
```
Comment (azizova-leyla) [inline app/models/pool.rb]: Is it smart to not do a query if index didn't change?

File: app/views/pools/pool_matches.html.erb [hunk 2]
```diff
@@ -1,14 +1,26 @@
...
-  <li><a href="<%= matches_show_path match.id %>">Match <%= index + 1 %></a>
-    <ul>
-      <% match.fighters.each do |fighter| %>
-      <li><%= fighter.first_name %> <%= fighter.last_name %></li>
-      <% end %>
-    </ul>
-  </li>
+<ul class="pool-matches-sortable">
+  <% @pool.matches.each do |match| %>
+    <li class="pool-matches-match" data-match-id="<%= match.id %>">
```
Comment (azizova-leyla) [inline app/views/pools/pool_matches.html.erb:10]: what does data-match-id do?

Comment (ytolstyk) [inline app/views/pools/pool_matches.html.erb:10]: This is a `data` attribute on an HTML tag, and I can grab this piece of data from JavaScript using `element.dataset.matchId`

File: app/models/pool.rb (hunk 1 above)
Comment (ytolstyk) [inline app/models/pool.rb]: Good point... I should probably handle this in JavaScript.

File: app/assets/javascripts/pools/pool_matches.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  },
+
+  listen: function() {
+    this.$saveButton.on('click', this.matchSaveClick);
+  },
+
+  matchSaveClick: function(event) {
+    event.preventDefault();
+
+    $.ajax({
```
Comment (ytolstyk) [inline app/assets/javascripts/pools/pool_matches.js:42]: This should only happen if `this.newPayload` is different from `this.oldPayload`.

**Pull request merged**